src/hlsBinaries.ts

```diff
@@ -364,6 +365,11 @@ export async function downloadHaskellLanguageServer(
     window.showInformationMessage(new NoBinariesError(releases[0].tag_name, ghcVersion).message);
     return null;
   }
+  if (release?.tag_name != releases[0].tag_name) {
+    const message = `haskell-language-server ${releases[0].tag_name} for GHC ${ghcVersion} is not available on ${os.type()}. Falling back to haskell-language-server ${release?.tag_name}`
+    logger.warn(message)
+    window.showInformationMessage(message)
```

Comment:
I am afraid this alert message will be shown each time hls is going to be used and that could be a little bit cumbersome
The problem is `downloadFile` is called each time you are gonna use hls but it does not actually download the file if it already exists.
So we should either check if the executable already exists and dont show the alert in that case (but keep the log warning) or make `downloadFile` return a Boolean or something which tell us if it has downloaded effectively the binary and only show the alert in that case.

Reply:
https://github.com/haskell/vscode-haskell/pull/506/commits/db4a6910d6ec96fac02cfd5b4c3c07e2f61c4f44 should improve this.

Reply:
great, thanks